src/ashley/machina_extensions/forum_permission/handler.py

```diff
@@ -43,3 +46,13 @@ def forum_list_filter(self, qs, user):
         if self.current_lti_context_id:
             return forums_to_show.filter(lti_contexts__id=self.current_lti_context_id)
         return forums_to_show
+
+    # pylint:disable = W0201
+    def _get_all_forums(self):
+        """ Returns all forums accessible for the LTIContext of the user. """
+        super()._get_all_forums()
+        if self.current_lti_context_id:
+            self._all_forums = Forum.objects.filter(
+                lti_contexts__id=self.current_lti_context_id
+            )
+        return self._all_forums
```

Comment:
In the original `_get_all_forums` method, there is a mechanism to avoid executing the request multiple time:
```python
        if not hasattr(self, '_all_forums'):
            self._all_forums = list(Forum.objects.all())
```

In your version, the request is executed everytime if `self.current_lti_context_id` is defined.

Reply:
true, I've added the same test